src/components/CommentList/index.spec.tsx

```diff
@@ -0,0 +1,207 @@
+import * as React from 'react';
+import { Store } from 'redux';
+
+import Comment from '../Comment';
+import configureStore from '../../configureStore';
+import {
+  ExternalComment,
+  actions as commentsActions,
+} from '../../reducers/comments';
+import {
+  createFakeExternalComment,
+  dispatchComments,
+  shallowUntilTarget,
+} from '../../test-helpers';
+
+import CommentList, { CommentListBase, PublicProps } from '.';
+
+describe(__filename, () => {
+  type RenderParams = Partial<PublicProps> & { store?: Store };
+
+  const render = ({
+    addonId = 1,
+    children = (content: JSX.Element) => content,
+    fileName = null,
+    line = null,
+    versionId = 2,
+    store = configureStore(),
+    ...moreProps
+  }: RenderParams = {}) => {
+    const props = {
+      addonId,
+      children,
+      fileName,
+      line,
+      versionId,
+      ...moreProps,
+    };
+    return shallowUntilTarget(<CommentList {...props} />, CommentListBase, {
+      shallowOptions: { context: { store } },
+    });
+  };
+
+  const createKeyParams = () => {
+    return { fileName: null, line: null, versionId: 1 };
+  };
+
+  const renderComments = ({
+    comments = [createFakeExternalComment()],
+    fileName,
+    line,
+    versionId,
+    ...props
+  }: RenderParams & { comments?: ExternalComment[] } = {}) => {
+    if (
+      fileName !== undefined ||
+      line !== undefined ||
+      versionId !== undefined
+    ) {
+      throw new Error('Defining custom key parameters is not supported');
```

Comment:
When I try to lazy-define the key params here, I get a TS error that doesn't make sense to me.

<details>
<summary>example</summary>

```
src/components/CommentList/index.spec.tsx:53:40 - error TS2345: Argument of type '{ comments: ExternalComment[]; fileName: string | null | undefined; line: number | null | undefined; versionId: number | undefined; }' is not assignable to parameter of type '{ store?: Store<ApplicationState, AnyAction> | undefined; comments?: { filename: string | null; id: number; lineno: number | null; canned_response: ExternalCannedResponse | null; comment: string | null; user: { ...; } | { ...; }; version: ExternalVersionWithContent; }[] | undefined; fileName?: null | undefined; line...'.
  Types of property 'fileName' are incompatible.
    Type 'string | null | undefined' is not assignable to type 'null | undefined'.
      Type 'string' is not assignable to type 'null | undefined'.

53     const { store } = dispatchComments({ ...finalKeyParams, comments });
```

</details>

It's not really important that this function accepts custom key params so I don't think it's worth tracking down the TS fix. I added the code to throw an error because someone might *think* they could specify custom key params. If they were to do so, then the render would work but the data could be missing. You might be staring at your test code thinking that you had the right key and wondering why it's still not working. So, I just want to throw an error.